src/iam/iam-base.ts

```diff
@@ -199,98 +195,127 @@ export class IAMBase {
-      this._didSigner = new IdentityOwner(
-        this._signer,
-        this._provider,
-        publicKey,
-        this._connectionOptions.privateKey
-      );
       return;
     }
-    if (this._runningInBrowser) {
-      const metamaskProvider = (await detectMetamask({
-        mustBeMetaMask: true
-      })) as any;
-      if (metamaskProvider && useMetamask) {
-        if (initializeMetamask) {
-          await metamaskProvider.request({
+    if (!this._runningInBrowser) {
+      throw new Error("Initialization of signer not possible in nodejs env without private key ");
+    }
+
+    const metamaskProvider: any = await detectMetamask({
+      mustBeMetaMask: true
+    });
+    if (metamaskProvider && useMetamask) {
+      const requestObject = initializeMetamask
+        ? {
             method: "wallet_requestPermissions",
             params: [
               {
                 eth_accounts: {}
               }
             ]
-          });
-        } else {
-          const accounts: string[] = await metamaskProvider.request({
+          }
+        : {
             method: "eth_accounts",
             params: [
               {
                 eth_accounts: {}
               }
             ]
-          });
-          if (accounts.length < 1) {
-            await metamaskProvider.request({
-              method: "wallet_requestPermissions",
-              params: [
-                {
-                  eth_accounts: {}
-                }
-              ]
-            });
-          }
-        }
-        const provider = new providers.Web3Provider(metamaskProvider);
-        this._signer = provider.getSigner();
-        const publicKey = await this.getPublicKey();
-        this._didSigner = new IdentityOwner(this._signer, this._provider, publicKey);
-        return;
+          };
+      const accounts: string[] = await metamaskProvider.request(requestObject);
+
+      if (!initializeMetamask && accounts.length < 1) {
+        await metamaskProvider.request({
+          method: "wallet_requestPermissions",
+          params: [
+            {
+              eth_accounts: {}
+            }
+          ]
+        });
       }
-      this._transactionOverrides = {
-        gasLimit: hexlify(4900000),
-        gasPrice: hexlify(0.1)
-      };
-      this._walletConnectProvider = new WalletConnectProvider({
-        rpc: {
-          [this._connectionOptions.chainId]: this._connectionOptions.rpcUrl
-        },
-        infuraId: this._connectionOptions.infuraId,
-        bridge: this._connectionOptions.bridgeUrl
-      });
-      await this._walletConnectProvider.enable();
-      this._signer = new providers.Web3Provider(this._walletConnectProvider).getSigner();
-      const publicKey = await this.getPublicKey();
-      this._didSigner = new IdentityOwner(this._signer, this._provider, publicKey);
+      const provider = new providers.Web3Provider(metamaskProvider);
+      this._signer = provider.getSigner();
```

Comment:
I find this part a bit confusing as I think the outcome is that `this._signer` is associated with the Web3Provider but then we pass a different (JsonRpc) provider into the `IdentityOwner` constructor. So `IdentityOwner` is then kind of associated with two different providers 🤔 ? I think that could be a source of confusion as to which network/chain "Switchboard" is actually connected to. Not super related to this PR and I don't have a great solution, just bringing it up for thought 😅